R/run_pacta.R

```diff
@@ -0,0 +1,28 @@
+#' Run PACTA for all portfolios matching a pattern in an input directory
+#'
+#' @param wd String. Working directory pointing to pacta's source code
+#' @param regext String. Regular expression that defines a valid portfolio name.
+#'
+#' @examples
+#' run_pacta()
+#' @noRd
+run_pacta <- function(wd = here::here()) {
+  withr::local_dir(wd)
+
+  portfolios <- portfolios()
+  command <- glue::glue("Rscript --vanilla web_tool_script_1.R {portfolios};\
+                          Rscript --vanilla web_tool_script_2.R {portfolios}")
+
+  for (i in seq_along(command)) {
+    message("Start portfolio: ", portfolios[[i]])
+    system(command[[i]])
+    message("End portfolio: ", portfolios[[i]])
+  }
+}
+
+#' Get the name of the portfolios
+portfolios <- function(path = "../input", regexp = "_Input[.]csv") {
+  csv <- fs::dir_ls(path, regexp = regexp)
+  fs::path_ext_remove(fs::path_file(csv))
+}
```

Comment:
In principle, I'm not a fan of having an R function that calls bash commands. This was implemented just because it makes looping over all input portfolios quite easy, but we could also just write a bash loop that does this.

Reply:
maybe use [withr](https://withr.r-lib.org)?

Reply:
also note that this commit 897433f71c18edb6ba2cf6798f4c961158f27b0d is super relevant here because it moves the stress testing to web_tool_script_3.R

Reply:
yup that's what made this possible :)

Reply:
Note to self: `R/run_pacta.R` and `R/source_web_tool_scripts.R` are basically the same, likely they can be combined into one function.

Reply:
I'm glad you recognize that this is a  problematic construction. Note that your current implementation starts an R session from within an R session. 🤯 

Instead, I would probably do this as a bash script, and have that the the CMD in the dockerfile.